File: Language/Go/design-patterns/behavioural-patterns/observer/src/subscriber.go
package src

import "fmt"

type Subscriber struct {
	Publisher    *Publisher
	Subscription int
}

func (s *Subscriber) Shutdown() {
	fmt.Println("Subscriber: Unsubscribing...")
	s.Publisher.Unsubscribe(s.Subscription)
}

func (s *Subscriber) Speak(d string) {
	fmt.Printf("Subscriber: Publisher says \"%s\"\n", d)
}

File: Language/Scala/design-patterns/behavioural-patterns/observer/src/main/scala/Subscriber.scala
package observer

class Subscriber(val publisher: Publisher) {
  var subscription: Option[Int] = None

  def shutdown(): Unit = {
    println("Subscriber: Unsubscribing...")
    publisher.unsubscribe(subscription)
  }

  def speak(d: String): Unit = println(s"Subscriber: Publisher says ${'"'}$d${'"'}")
}
